sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/generators.scala

```diff
@@ -222,6 +222,51 @@ case class Stack(children: Seq[Expression]) extends Generator {
   }
 }
 
+/**
+ * Replicate the row N times. N is specified as the first argument to the function.
+ * {{{
+ *   SELECT replicate_rows(2, "val1", "val2") ->
+ *   2  val1  val2
+ *   2  val1  val2
+ *  }}}
+ */
+@ExpressionDescription(
+usage = "_FUNC_(n, expr1, ..., exprk) - Replicates `n`, `expr1`, ..., `exprk` into `n` rows.",
```

Comment:
I checked the design doc for INTERSECT ALL and EXCEPT ALL. Looks like the `n` is always stripped after Generate operation. So why we need to keep `n` in  `ReplicateRows` outputs? Can we do it like:

```
> SELECT _FUNC_(2, "val1", "val2");
  val1  val2
  val1  val2
```

Reply:
@viirya I did think about it Simon. But then, i decided to match the output with Hive.